notify/rocketchat/rocketchat.go

```diff
@@ -0,0 +1,272 @@
+		if err != nil {
+			return false, err
+		}
+		level.Warn(n.logger).Log("msg", "Truncated title", "key", key, "max_runes", maxTitleLenRunes)
+	}
+	att := &Attachment{
+		Title:     title,
+		TitleLink: tmplText(n.conf.TitleLink),
+		Text:      tmplText(n.conf.Text),
+		ImageURL:  tmplText(n.conf.ImageURL),
+		ThumbURL:  tmplText(n.conf.ThumbURL),
+		Color:     tmplText(n.conf.Color),
+	}
+	numFields := len(n.conf.Fields)
+	if numFields > 0 {
+		fields := make([]config.RocketchatAttachmentField, numFields)
+		for index, field := range n.conf.Fields {
+			// Check if short was defined for the field otherwise fallback to the global setting
+			var short bool
+			if field.Short != nil {
+				short = *field.Short
+			} else {
+				short = n.conf.ShortFields
+			}
+
+			// Rebuild the field by executing any templates and setting the new value for short
+			fields[index] = config.RocketchatAttachmentField{
+				Title: tmplText(field.Title),
+				Value: tmplText(field.Value),
+				Short: &short,
+			}
+		}
+		att.Fields = fields
+	}
+	numActions := len(n.conf.Actions)
+	if numActions > 0 {
+		actions := make([]config.RocketchatAttachmentAction, numActions)
+		for index, action := range n.conf.Actions {
+			rocketchatAction := config.RocketchatAttachmentAction{
+				Type: "button", // Only button type is supported
+				Text: tmplText(action.Text),
+				URL:  tmplText(action.URL),
+				Msg:  tmplText(action.Msg),
+			}
+
+			actions[index] = rocketchatAction
+		}
+		att.Actions = actions
+	}
+
+	body := &PostMessage{
+		Channel:     n.conf.Channel,
+		Emoji:       tmplText(n.conf.Emoji),
+		Avatar:      tmplText(n.conf.IconURL),
+		Attachments: []Attachment{*att},
+	}
+	if err != nil {
+		return false, err
+	}
+
+	var buf bytes.Buffer
+	if err := json.NewEncoder(&buf).Encode(body); err != nil {
+		return false, err
+	}
+	url := fmt.Sprintf("%s/%s", n.conf.APIURL.String(), "api/v1/chat.postMessage")
+	resp, err := n.postJSONFunc(ctx, n.client, url, &buf)
+	if err != nil {
+		return true, notify.RedactURL(err)
+	}
+	defer notify.Drain(resp)
+
+	// Use a retrier to generate an error message for non-200 responses and
+	// classify them as retriable or not.
+	retry, err := n.retrier.Check(resp.StatusCode, resp.Body)
+	if err != nil {
+		err = fmt.Errorf("%w: channel %q", err, body.Channel)
+		return retry, notify.NewErrorWithReason(notify.GetFailureReasonFromStatusCode(resp.StatusCode), err)
+	}
+
+	// Rocketchat web API might return errors with a 200 response code.
+	retry, err = checkResponseError(resp)
+	if err != nil {
+		err = fmt.Errorf("%w: channel %q", err, body.Channel)
+		return retry, notify.NewErrorWithReason(notify.ClientErrorReason, err)
+	}
+
+	return retry, nil
+}
+
+// checkResponseError parses out the error message from Rocketchat API response.
+func checkResponseError(resp *http.Response) (bool, error) {
+	body, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return true, fmt.Errorf("%w: could not read response body: ", err)
```

Comment:
In Go it's idiomatic to wrap the error at the end:

```suggestion
		return true, fmt.Errorf("could not read response body: %w", err)
```